src/gt4py/next/iterator/transforms/infer_domain.py

```diff
@@ -129,6 +145,7 @@ def infer_as_fieldop(
     applied_fieldop: itir.FunCall,
     target_domain: DOMAIN,
     offset_provider: common.OffsetProvider,
+    symbolic_domain_sizes: Optional[dict[str, str]],
```

Comment:
Needs explanation

Reply:
Added a docstring to `infer_expr` (and pass manager) not here and made all `infer_<node type>` methods private so that one looks for the documentation in the public api not here.